### PR TITLE
:lipstick: :technologist: :mortar_board: `literalinclude` --- Use tags for better referencing of code

### LINKED ISSUE #479: 💄 🔭 Use `literalinclude` rst directive
### What 
Looks like, instead of copy/pasting code into an rst file, [you can reference an actual chunk of code from another file](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-literalinclude), like maybe the actual `.java` file. **However**, it's not clear how great this would be with java vs python.

This may seem more good than it is at this stage as there are some deliberate differences between the code in the rst files and the actual code. Either way, this is something for futur consideration.

### Why
I can see this helping with consistency. 

### Where
Everywhere. Including cs101 too.

### PR DESCRIPTION
**Pull request by jameshughes89**
### Related Issues or PRs
Closes #479 

### What
Add tags to relevant bits of code and replace all the line numbers in the `literalinclude` with tags

### Why
code moves, line numbers get messed up, braking the website 

### How
Slowly

### Testing
:+1: 
I did one block at a time. It was... painful... 

### Additional Notes
Noticed #943 along the way.

### CODE REVIEW

File: src/site/topics/algorithm-analysis/algorithm-analysis.rst [hunk 1]
```diff
@@ -386,21 +386,21 @@ Stack Comparisons
...
+Popping & Peeking
+-----------------
 
 .. literalinclude:: /../main/java/LinkedStack.java
     :language: java
     :linenos:
-    :lineno-start: 32
-    :lines: 32-41
+    :start-after: // [start-pop_peek]
+    :end-before: // [end-pop_peek]
```
Comment (twentylemon) [inline src/site/topics/algorithm-analysis/algorithm-analysis.rst:396]: Are the `//` necessary? Only `[tag]` seems more natural.

**Review by twentylemon — APPROVED** [review]
Maybe there's rst plugins or something that can extract a single method? Just thinking aloud, I have no idea if anything like that exists.

File: src/site/topics/algorithm-analysis/algorithm-analysis.rst (hunk 1 above)
Comment (jameshughes89) [reply src/site/topics/algorithm-analysis/algorithm-analysis.rst:396]: agreed --- will check tomorrow

**Comment by jameshughes89** [comment]
> Maybe there's rst plugins or something that can extract a single method? Just thinking aloud, I have no idea if anything like that exists.

Maybe there is a plugin, but base RST only withs with "pyobjects" which is lame.

**Pull request merged**